x/kyllian/dust-mcp/src/cli.ts

```diff
@@ -1,31 +1,135 @@
+#!/usr/bin/env node
+
 import { Command } from "commander";
-import { authenticate } from "./auth.js";
-import { isAuthenticated } from "./config.js";
-
-const program = new Command();
-
-program.name("dust-mcp").description("Dust MCP CLI").version("1.0.0");
-
-program
-  .command("auth")
-  .description("Authenticate with Dust")
-  .action(async () => {
-    try {
-      // const alreadyAuthenticated = await isAuthenticated();
-      // if (alreadyAuthenticated) {
-      //   console.log("You are already authenticated.");
-      //   return;
-      // }
-
-      console.log("Authenticating with Dust...");
-      await authenticate();
-      console.log(
-        "Authentication successful! Your credentials have been saved."
-      );
-    } catch (error) {
-      console.error("Authentication failed:", error);
+import { getDustAPI } from "./api.js";
+import { isLoggedIn, login, logout } from "./auth.js";
+import { startServer } from "./server.js";
+import { version } from "./version.js";
+
+const loginAction = async (quiet = false) => {
+  try {
+    const isUserLoggedIn = await isLoggedIn();
+    if (isUserLoggedIn) {
+      if (!quiet) console.log("You are already logged in.");
+      return;
+    }
+
+    if (!quiet) console.log("Logging in to Dust...");
+    await login();
+    if (!quiet)
+      console.log("Login successful! Your credentials have been saved.");
+  } catch (error) {
+    console.error("Login failed:", error);
+    process.exit(1);
+  }
+};
+
+const logoutAction = async () => {
+  try {
+    const isUserLoggedIn = await isLoggedIn();
+    if (!isUserLoggedIn) {
+      console.log("You are not currently logged in.");
+      return;
+    }
+
+    await logout();
+    console.log("Successfully logged out.");
+  } catch (error) {
+    console.error("Logout failed:", error);
+    process.exit(1);
+  }
+};
+
+const statusAction = async () => {
+  try {
+    const isUserLoggedIn = await isLoggedIn();
+    if (!isUserLoggedIn) {
+      console.log("You are not currently logged in.");
+      return;
+    }
+
+    const { dustAPI, dustConfig } = await getDustAPI();
+    const userRes = await dustAPI.me();
+
+    if (!userRes.isOk()) {
+      console.error("Failed to get user information:", userRes.error.message);
       process.exit(1);
     }
-  });
 
-program.parse(process.argv);
+    const user = userRes.value;
+    const currentWorkspace = user.workspaces.find(
+      (w) => w.sId === dustConfig.workspaceId
+    );
+
+    console.log("Currently logged in as:");
+    console.log(`  ${user.fullName} <${user.email}>`);
+
+    if (!currentWorkspace) {
+      console.error("Failed to find current workspace in user workspaces");
+      process.exit(1);
+    }
+
+    console.log(
+      `  - Workspace: ${currentWorkspace.name} [${currentWorkspace.sId}]`
+    );
+  } catch (error) {
+    console.error("Failed to get status:", error);
+    process.exit(1);
+  }
+};
+
+const switchAction = async () => {
```

Comment:
I think this might be over-complicating a bit, calling login again should be enough to sign in to a different account ?

Reply:
(loosely help opinion though)